File: pcme-platform/packages/lab-client/src/types/event.ts
export interface ExperimentEvent {
  seq: number;
  ts_abs: number; // absolute timestamp in ms (Date.now())
  ts_video: number; // video currentTime in seconds
  event_type: string;
  payload: Record<string, unknown>;
  source: string;
}

export type EventType =
  | "video_state_change"
  | "video_seeked"
  | "video_time_update"
  | "recording_start"
  | "recording_stop"
  | "physio_sync"
  | "session_start"
  | "session_end"
  | "questionnaire_submit"
  | "custom";

File: pcme-platform/server/app/schemas/event.py
from pydantic import BaseModel


class EventItem(BaseModel):
    seq: int
    ts_abs: int  # absolute timestamp ms
    ts_video: float  # video currentTime in seconds
    event_type: str
    payload: dict
    source: str


class EventBatchUpload(BaseModel):
    events: list[EventItem]


class EventBatchResponse(BaseModel):
    batch_id: str
    event_count: int
    log_file_path: str
